src/common/signalk/SKNMEA2000Parser.cpp

```diff
@@ -80,42 +128,207 @@ const SKUpdate& SKNMEA2000Parser::parse128259(const SKSourceInput& input, const
       }
-      else if (offset > 0) {
+      else if (offset >= 0) {
         update->setEnvironmentDepthSurfaceToTransducer(offset);
         update->setEnvironmentDepthBelowSurface(depthBelowTransducer + offset);
       }
+
+      _sku = update;
+      return *_sku;
+    }
+  }
+  DEBUG("Unable to parse N2kMsg with PGN %i", msg.PGN);
+  return _invalidSku;
+}
+
+// *****************************************************************************
+//   PGN 130306 W I N D
+// The boat referenced true wind is given by the vector sum of Apparent wind and vessel's heading and speed though the water.
+// The ground referenced true wind is given by the vector sum of Apparent wind and vessel's heading and speed over ground.
+// References and calculations : https://www.rocktheboatmarinestereo.com/specs/MSNRX200I.pdf
+//    tN2kWindReference:
+//      N2kWind_True_North=0    => Ground Wind, calculated using SOG/COG, referenced to true north
+//      N2kWind_Magnetic=1      => Ground Wind, calculated by SOG/COG, referenced to magnetic north
+//      N2kWind_Apparent=2      => Apparent Wind, measured on boat, relative to vessel centerline
+//      N2kWind_True_boat=3     => Ground Wind, calculated using SOG/COG, relative to centerline
+//      N2kWind_True_water=4    => Theoretical Wind, calc using Heading/STW, relative to centerline vessel, referenced to water
+// *****************************************************************************
+const SKUpdate& SKNMEA2000Parser::parse130306(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
+  unsigned char sid;
+  double windSpeed = N2kDoubleNA;     // m/s
+  double windAngle = N2kDoubleNA;     // in Rad
+  tN2kWindReference windReference;
+
+  if (ParseN2kPGN130306(msg,sid,windSpeed,windAngle,windReference)) {
+    SKUpdateStatic<2> *update = new SKUpdateStatic<2>();
+    update->setTimestamp(timestamp);
+
+    SKSource source = SKSource::sourceForNMEA2000(input, msg.PGN, msg.Priority, msg.Source);
+    update->setSource(source);
+
+    if (!N2kIsNA(windAngle) && !N2kIsNA(windSpeed)) {
+      if (windReference == N2kWind_True_North) {
+        // Ground Wind Speed
+        update->setEnvironmentWindSpeedOverGround(windSpeed);
+        // Ground Wind Direction
+        update->setEnvironmentWindDirectionTrue(windAngle);
+      }
+      else if (windReference == N2kWind_Magnetic) {
+        // Ground Wind Speed
+        update->setEnvironmentWindSpeedOverGround(windSpeed);
+        // Ground Wind Direction refered to magnetic north
+        update->setEnvironmentWindDirectionMagnetic(windAngle);
+      }
+      //TODO: correct if Timos Typo will be corrected in library
+      else if (windReference == N2kWind_Apprent) {
+        // AWS Apparent Wind Speed
+        update->setEnvironmentWindSpeedApparent(windSpeed);
+        // AWA pos coming from starboard, neg from port, relative to centerline vessel
+        if (windAngle > M_PI) windAngle *= -1;
```

Comment:
At first I thought this was ok but now I do not think this works. If you think about it in degrees, if the angle is 185 (slightly more than PI), then -1*185 will be -185 degrees which is 175 degrees. You have actually changed the value. Not good.

Maybe we should have a function (or a macro) to normalize an angle between -M_PI and +M_PI. 
Something like `#define NORMALIZE_ANGLE(x) fmod(x, 2*M_PI)-M_PI`. This way we do not risk repeating the same error and it will make the code more readable.

Reply:
You are right of course!
The normalize macro is a great idea, where to put it best?
SKUnits.h ?